cmds/report.c

```diff
@@ -185,6 +185,39 @@ static void print_and_delete(struct rb_root *root, bool sorted, void *arg,
 	}
 }
 
+static void print_csv_function(struct uftrace_report_node *node, void *unused)
+{
+	if (avg_mode == AVG_NONE) {
+		print_time_unit(node->total.sum);
```

Comment:
Hmm.. now I think that we can just print the raw number for CSV as it's more machine-friendly IMO.

Reply:
What about this?
```diff
diff --git a/cmds/report.c b/cmds/report.c
index 4b25f53..a81da01 100644
--- a/cmds/report.c
+++ b/cmds/report.c
@@ -188,11 +188,9 @@ static void print_and_delete(struct rb_root *root, bool sorted, void *arg,
 static void print_csv_function(struct uftrace_report_node *node, void *unused)
 {
        if (avg_mode == AVG_NONE) {
-               print_time_unit(node->total.sum);
-               pr_out(",");
-               print_time_unit(node->self.sum);
-               pr_out(",");
-               pr_out("%lu,%s\n", node->call, node->name);
+               pr_out("%"PRIu64", ", node->total.sum);
+               pr_out("%"PRIu64", ", node->self.sum);
+               pr_out("%lu, %s\n", node->call, node->name);
        }
        else {
                uint64_t time_avg, time_min, time_max;
@@ -207,12 +205,9 @@ static void print_csv_function(struct uftrace_report_node *node, void *unused)
                        time_min = node->self.min;
                        time_max = node->self.max;
                }
-               print_time_unit(time_avg);
-               pr_out(",");
-               print_time_unit(time_min);
-               pr_out(",");
-               print_time_unit(time_max);
-               pr_out(",");
+               pr_out("%"PRIu64", ", time_avg);
+               pr_out("%"PRIu64", ", time_min);
+               pr_out("%"PRIu64", ", time_max);
                pr_out("%s\n", node->name);
        }
 }
@@ -255,7 +250,7 @@ static void report_functions(struct uftrace_data *handle, struct opts *opts)
        struct rb_root name_root = RB_ROOT;
        struct rb_root sort_root = RB_ROOT;
        const char f_format[] = "  %10.10s  %10.10s  %10.10s  %-.*s\n";
-       const char csv_format[] = "%s,%s,%s,%s\n";
+       const char csv_format[] = "%s, %s, %s, %s\n";
        const char line[] = "=================================================";
 
        build_function_tree(handle, &name_root, opts);
```
The output looks as follows:
```
$ uftrace report --format=csv
Total time, Self time, Calls, Function
53488347, 6688737, 1, Py_Initialize
40379422, 23148307, 26, PyEval_EvalCode
38906171, 977127, 4, PyImport_ImportModule
3441877, 3441877, 7106, strlen
2986358, 2986358, 10517, memcpy
2624052, 2572081, 4746, _PyObject_GC_Malloc
2118975, 2118975, 7560, memset
2071407, 2071407, 2733, strcmp
1853438, 1853438, 5900, memcmp
    ...
```

Reply:
But I'm not sure if it's better to show raw number as this CSV is a kind of human friendly format for excel users.

Reply:
No I think CSV is a machine-friendly format to draw graphs.

Reply:
@namhyung Sorry. I didn't understand.
What does machine-friendly format mean?

Reply:
Nevermind, I think that CSV will be useful when someone wants to draw some graph using the report output.  So we don't need to care much to make the output human-readable.  I think it's easier to handle by Excel-like programs if we use raw number (without units like ms).